fix flushBackoffQCompleted: remove defer

duc-cnzj opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind bug

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
resource leak.

#### Which issue(s) this PR fixes:

fix resource leak in loop.

https://github.com/kubernetes/kubernetes/blob/3f6793de4862647fc221f4acf9d61adace73a592/pkg/scheduler/internal/queue/scheduling_queue.go#L410-L433

remove defer

https://github.com/kubernetes/kubernetes/blob/3f6793de4862647fc221f4acf9d61adace73a592/pkg/scheduler/internal/queue/scheduling_queue.go#L431

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

duc-cnzj committed:
fix flushBackoffQCompleted: remove defer

'defer' is called in the 'for' loop, remove it

ahg-g commented on pkg/scheduler/internal/queue/scheduling_queue.go:
```diff
@@ -428,7 +428,7 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 		p.activeQ.Add(rawPodInfo)
 		metrics.SchedulerQueueIncomingPods.WithLabelValues("active", BackoffComplete).Inc()
-		defer p.cond.Broadcast()
+		p.cond.Broadcast()
```
I would change the returns above to breaks, add a flag to indicate whether or not to broadcast and move the broadcast call outside the loop.

DuC-cnZj replied:
@ahg-g  Thanks for the tip, updated

duc-cnzj committed:
a flag to indicate whether or not to broadcast

change the returns above to breaks, add a flag to indicate whether or not to broadcast.

ahg-g commented:
/lgtm
/approve

denkensk commented:
/retest-required

kerthcet commented:
/release-note-none

kerthcet commented:
/retest

kerthcet commented:
/retest

binacs reviewed (approved):
/ok-to-test
